### DISCUSSION TITLE
Serving conent of a temporary file which is an injected dependency stopped working in v 0.106.0

### DISCUSSION THREAD

**Discussion by mezhaka** [fastapi/fastapi#10850] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import tempfile
from typing import Annotated

import fastapi
from fastapi import status, responses, testclient


def tmp_file() -> tempfile.NamedTemporaryFile():
    with tempfile.NamedTemporaryFile() as tmp:
        yield tmp


app = fastapi.FastAPI()


@app.get("/test", response_class=responses.FileResponse, status_code=status.HTTP_200_OK)
def serve(
    tmp_file: Annotated[tempfile.NamedTemporaryFile, fastapi.Depends(tmp_file)],
):
    tmp_file.write(b"test")
    tmp_file.seek(0)
    return responses.FileResponse(tmp_file.name)


@app.get("/test-no-inject", response_class=responses.FileResponse, status_code=status.HTTP_200_OK)
def serve_no_inject():
    def stream():
        with tempfile.NamedTemporaryFile() as tmp_file:
            tmp_file.write(b"test")
            tmp_file.seek(0)
            yield from tmp_file
    return responses.StreamingResponse(stream())


def test_serve():
    with testclient.TestClient(app) as client_:
        r = client_.get("/test")
    assert r.content == b"test"


def test_serve_no_inject():
    with testclient.TestClient(app) as client_:
        r = client_.get("/test-no-inject")
    assert r.content == b"test"
```

### Description

`test_serve` used to work fine in v 0.105.0, but gives an exception in 0.106.0:
```
    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
        if self.stat_result is None:
            try:
                stat_result = await anyio.to_thread.run_sync(os.stat, self.path)
                self.set_stat_headers(stat_result)
            except FileNotFoundError:
>               raise RuntimeError(f"File at path {self.path} does not exist.")
E               RuntimeError: File at path /var/folders/gs/bdkyw8_x60sc_f1zss1gc30c0000gp/T/tmpv1x9vqb0 does not exist.
```

I suspect it has to do with the breaking change in 0.106.0 and would like to get a suggestion what is the idiomatic way of serving a temporary file.

The technique I used is also suggested in this SO discussion: https://stackoverflow.com/a/72535413/18775, but it seems this way is broken now.

There is a discussion here https://github.com/tiangolo/fastapi/discussions/8633 where a StreamingResponse is used--in my example `test-no-inject` endpoint works fine and `test_serve_no_inject` demonstrates it, however it is prone to an file descriptor leak in case response is broken halfway.

And here https://github.com/tiangolo/fastapi/issues/2152 a background task is used to delete a temporary file.

I am in search for an idiomatic way, which is going to decrease the chances I have to rework this bit again.

### Operating System

macOS

### Operating System Details

I also tried it on Linux

### FastAPI Version

0.106.0

### Pydantic Version

1.10.13

### Python Version

Python 3.10.13

**Comment by rushilsrivastava** [comment]
The only viable solution I can think of at the moment is using `BackgroundTasks`, as suggested in #2512. This feels like a downgrade to the behavior of dependencies, but I understand there are other advantages of having dependencies resolve before the response is returned to the client.

Related issue: #10857

**Comment by etienne-monier** [comment]
Hi ! I've got the same issue ! I need to stay below 0.106.0
